azure-iot-device/azure/iot/device/iothub/aio/async_clients.py

```diff
@@ -130,6 +130,38 @@ async def _disable_feature(self, feature_name):
             # This branch shouldn't be reached, but in case it is, log it
             logger.info("Feature ({}) already disabled - skipping".format(feature_name))
 
+    def _generic_receive_handler_setter(self, handler_name, feature_name, new_handler):
```

Comment:
Something to think about:  I think it was maybe overkill to support both coroutines and functions as handlers, but another way to do this would be to wrap the function inside a coroutine inside _generic_receive_handler_setter.  This way, for async, all handlers are coroutines, and some of them internally call a function (on a different thread, of course, like we do.)

Reply:
We have to support coroutines because of the handlers that need to invoke client coroutine methods (e.g. `.send_method_response()` )